wordpress-mobile/WordPress-Android
# [Blogging Prompts Enhancements] Turn off prompt card menu item

Part of #17803

Add "Turn off prompts" option to Prompt card overflow menu. Hitting it will turn off the feature (disable the Blogging Prompts setting inside Site Settings) and show a snackbar telling the user where to enable it again + option to undo and turn it back on.

This feature is hidden behind the BP Enhancements Feature Flag, so the new option should only be visible with the FF on.

![image](https://user-images.githubusercontent.com/5091503/216332220-aded9e33-d911-4e72-8685-ac9facaca018.png)

![image](https://user-images.githubusercontent.com/5091503/216433872-f90a9581-cef2-4200-ab7f-eef66a845815.png)

Removed the `Needs Design Review` as @paulforgione confirmed the correct copies to be used and I updated it in the code.

## To test:
Be sure to log in with a user and site that is a blog (can see the Prompts feature)

### Menu item appears only with FF on
1. Install Jetpack and log in
2. Open "Debug settings" and make sure `BloggingPromptsEnhancementsFeatureConfig` is disabled
3. Open "My Site" -> "HOME"
4. **Verify** the `Blogging Prompts` card is showing
5. Tap the overflow menu on the top-right of the card
7. **Verify** the `Turn off prompts` item is NOT available
8. Go back
9. Open "Debug settings" (avatar on top right -> App Settings -> Debug Settings)
10. Enable `BloggingPromptsEnhancementsFeatureConfig` in "Features in Development" section
11. Tap the "Restart the app" button
3. Open "My Site" -> "HOME"
4. **Verify** the `Blogging Prompts` card is showing
5. Tap the overflow menu on the top-right of the card
7. **Verify** the `Turn off prompts` item IS available

### "Turn off prompts" work as expected
1. Install Jetpack and log in
2. Make sure `BloggingPromptsEnhancementsFeatureConfig` is ENABLED
3. Open "My Site" -> "MENU"
4. Tap `Site Settings`
5. **Verify** the `Blogging Prompts` setting is turned ON
6. Open "My Site" -> "HOME"
5. **Verify** the `Blogging Prompts` card is showing
6. Tap the FAB
7. **Verify** the prompt action shows up in the bottom sheet
8. Close the bottom sheet
9. Tap the overflow menu on the top-right of the Prompts card
10. Tap the `Turn off prompts` item
11. **Verify** the card disappears
12. **Verify** the correct snackbar with an UNDO action is shown
13. Once the snackbar disappears, tap the FAB
14. **Verify** the prompt action DOES NOT show up in the bottom sheet
15. Open "My Site" -> "MENU"
16. Tap `Site Settings`
17. **Verify** the `Blogging Prompts` setting is turned off
18. Turn the `Bogging Prompt` setting on again
19. Do steps **6-14** again
20. Tap the UNDO action in the snackbar
21. **Verify** the feature is still turned on everywhere again (**steps 3-9**)

## Regression Notes
1. Potential unintended areas of impact
Incorrect loading of dashboard, bottom sheet, and settings.

22. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual and unit testing.

23. What automated tests I added (or what prevented me from doing so)
Updated and added new unit tests.

PR submission checklist:
